Formula/ecl.rb

```diff
@@ -14,19 +14,39 @@ class Ecl < Formula
   end
 
   depends_on "gmp"
+  depends_on "bdw-gc"
 
   def install
-    system "./configure", "--prefix=#{prefix}",
+    # These environment variables are necessary or gmp / boehmgc won't be found
+    # No idea why
+    gmp = Formula["gmp"]
+    boehmgc = Formula["boehmgc"]
+    ENV.append "CFLAGS", "-I#{gmp.opt_include} -I#{boehmgc.opt_include}"
+    ENV.append "CXXFLAGS", "-I#{gmp.opt_include} -I#{boehmgc.opt_include}"
+    ENV.append "LDFLAGS", "-L#{gmp.opt_lib} -L#{boehmgc.opt_lib}"
+
+    # https://gitlab.com/embeddable-common-lisp/ecl/tree/revert-make-split
+    ENV.deparallelize
+
+    system "./configure",
+                          "--prefix=#{prefix}",
                           "--enable-threads=yes",
-                          "--with-system-gmp=yes"
+                          "--with-dffi=included",
+                          "--with-cxx",
+                          "--enable-gmp=system"
     system "make"
     system "make", "install"
+    system "make", "check"
```

Comment:
The only point I can see is that it would be better to improve the formula so that the checks can fail, rather than omit them. But I haven's figured out how to make it work. When it runs it wants to find its own libraries at their installed location, but something about the build environment prevents it from finding the libraries, even though the libraries have been installed by `make install`. Maybe some more editing of environment variables (like `LD_LIBRARY_PATH` could get it working, but you've already expressed disfavor towards that general approach.